app/test_main.py

```diff
@@ -1 +1,53 @@
-# write your code here
+import pytest
+from unittest import mock
+
+
+from app.main import can_access_google_page
+
+
+@pytest.fixture()
+def mocked_valid_google_url() -> None:
+    with mock.patch("app.main.valid_google_url") as mocked_func:
+        yield mocked_func
+
+
+@pytest.fixture()
+def mocked_has_connection() -> None:
+    with mock.patch("app.main.has_internet_connection") as mocked_func:
+        yield mocked_func
+
+
+def test_valid_connection_and_url_exist(mocked_valid_google_url: any,
+                                        mocked_has_connection: any
```

Comment:
Your mocked_valid_google_url and mocked_has_connection fixtures both should yield mocks, right? so why dont you type hint it in your tests like "mock" instead of "any"?

Reply:
Yep, that is good one: fixed) Thanks)

Reply:
what about other test functions? im afraid i cant see it was changed anywhere beside this test :) just write me back here in case you've already done it and im missing something.

Reply:
it is still not fixed in all other similar cases

Reply:
Fixed)